test/unit/spec/default_spec.rb

```diff
@@ -0,0 +1,305 @@
+
+require_relative 'spec_helper'
+include CernerSplunk::PathHelpers
+
+platform_urls = {
+  redhat: {
+    '7.1': {
+      splunk: 'https://download.splunk.com/products/splunk/releases/6.3.4/linux/splunk-6.3.4-cae2458f4aef-linux-2.6-x86_64.rpm',
+      universal_forwarder: 'https://download.splunk.com/products/universalforwarder/releases/6.3.4/linux/splunkforwarder-6.3.4-cae2458f4aef-linux-2.6-x86_64.rpm'
+    }
+  },
+  ubuntu: {
+    '16.04': {
+      splunk: 'https://download.splunk.com/products/splunk/releases/6.3.4/linux/splunk-6.3.4-cae2458f4aef-linux-2.6-amd64.deb',
+      universal_forwarder: 'https://download.splunk.com/products/universalforwarder/releases/6.3.4/linux/splunkforwarder-6.3.4-cae2458f4aef-linux-2.6-amd64.deb'
+    }
+  },
+  windows: {
+    '2012R2': {
+      splunk: 'https://download.splunk.com/products/splunk/releases/6.3.4/windows/splunk-6.3.4-cae2458f4aef-x64-release.msi',
+      universal_forwarder: 'https://download.splunk.com/products/universalforwarder/releases/6.3.4/windows/splunkforwarder-6.3.4-cae2458f4aef-x64-release.msi'
+    }
+  },
+  suse: { # Generic Linux
+    '12.0': {
+      splunk: 'https://download.splunk.com/products/splunk/releases/6.3.4/linux/splunk-6.3.4-cae2458f4aef-Linux-x86_64.tgz',
+      universal_forwarder: 'https://download.splunk.com/products/universalforwarder/releases/6.3.4/linux/splunkforwarder-6.3.4-cae2458f4aef-Linux-x86_64.tgz'
+    }
+  }
+}
+
+describe 'splunk_install' do
+  let(:test_params) { {} }
+  let(:platform) { 'redhat' }
+  let(:version) { '7.1' }
+  let(:do_before) {}
+  let(:run_state) { {} }
+  let(:chef_runner) do
+    parameters = { step_into: ['splunk_install'], platform: platform, version: version }
+    ChefSpec::SoloRunner.new(parameters) do |node|
+      node.set['test_parameters'] = test_params
+      node.set['run_state'].merge!(run_state)
+    end
+  end
+  let(:chef_run) do
+    do_before
+    chef_runner.converge('cerner_splunk_ingredient_test::unit_test')
+  end
+  subject { chef_run }
+
+  describe 'action :install' do
+    platform_urls.each do |platform, versions|
+      versions.each do |version, urls|
+        context "on #{platform} #{version}" do
+          let(:platform) { platform.to_s }
+          let(:version) { version.to_s }
+          context 'with the Splunk package' do
+            let(:test_params) { { build: 'cae2458f4aef', version: '6.3.4', accept_license: true } }
+            let(:package_dir) { "./test/unit/.cache/#{filename_from_url(urls[:splunk])}" }
+
+            it 'should set the run state' do
+              expect(chef_run.node.run_state['splunk_ingredient']['current_installation']).to eq(
+                name: 'splunk',
+                package: :splunk,
+                version: '6.3.4',
+                build: 'cae2458f4aef',
+                x64: true
+              )
+            end
+
+            it { is_expected.to create_remote_file(package_dir).with(source: urls[:splunk]) }
+
+            case platform
```

Comment:
At the very least you can have a shared example for installation of packages that relies on platform, and any parameters used in the install block. I can see that kind of thing all over this file. In addition, you can have shared contexts for things such as your contexts around "with the Splunk package" and "with the Universal Forwarder package". There are lots of ways we can clean this up.